### PR TITLE
Pool Roles and actions

### PR DESCRIPTION
**Pull request by vedhavyas**
Spec: https://centrifuge.hackmd.io/Lga8cxI9Tia-u0ZPXrt84g

### CODE REVIEW

File: pallets/loan/src/mock.rs [hunk 1]
```diff
@@ -224,6 +231,14 @@ impl Default for TestExternalitiesBuilder {
 	}
 }
 
+parameter_types! {
+	pub const PoolAdmin: u64 = 1;
+	pub const Borrower: u64 = 2;
+	pub const RiskAdmin: u64 = 3;
+	pub const DropInvestor: u64 = 4;
```
Comment (offerijns) [inline pallets/loan/src/mock.rs:238]: We're not going to be using the DROP/TIN token names anymore for TOCC as we'll have multiple tranches. It's probably fine in the mocks/tests, but ideally we'd just call it senior/junior investors.

File: libs/common-traits/src/lib.rs [hunk 2]
```diff
@@ -103,26 +106,37 @@ pub trait PoolNAV<PoolId, Amount> {
...
-/// A trait that support pool reserve operations such as borrow and repay
-pub trait PoolReserve<Origin, AccountId> {
+/// Role can hold any type of role specific functions a user can do on a given pool.
+#[derive(Encode, Decode, Clone, Copy, PartialEq)]
+#[cfg_attr(any(feature = "std", feature = "runtime-benchmarks"), derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role {
+	PoolAdmin,
+	Borrower,
+	PricingAdmin,
```
Comment (offerijns) [inline libs/common-traits/src/lib.rs:116]: This has a `PricingAdmin` while the spec doesn't, is that to separate issue from activate/price permissions?

**Review by offerijns — COMMENTED** [review]
That was quick! 🔥

File: libs/common-traits/src/lib.rs (hunk 2 above)
Comment (vedhavyas) [reply libs/common-traits/src/lib.rs:116]: yeap. It would better to keep them seperate though

File: pallets/loan/src/mock.rs (hunk 1 above)
Comment (vedhavyas) [reply pallets/loan/src/mock.rs:238]: true. Will change this

File: libs/common-traits/src/lib.rs [hunk 3]
```diff
@@ -103,26 +106,37 @@ pub trait PoolNAV<PoolId, Amount> {
 	fn update_nav(pool_id: PoolId) -> Result<Amount, DispatchError>;
 }
 
-/// A trait that support pool reserve operations such as borrow and repay
-pub trait PoolReserve<Origin, AccountId> {
+/// Role can hold any type of role specific functions a user can do on a given pool.
+#[derive(Encode, Decode, Clone, Copy, PartialEq)]
+#[cfg_attr(any(feature = "std", feature = "runtime-benchmarks"), derive(Debug))]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum Role {
```
Comment (mustermeiszer) [inline libs/common-traits/src/lib.rs:113]: Maybe name `PoolRole` or similiar? Or are we expecting `Role` to be have roles not associated with a pool?

**Review by mustermeiszer — APPROVED** [review]
LGTM!

File: libs/common-traits/src/lib.rs (hunk 3 above)
Comment (vedhavyas) [reply libs/common-traits/src/lib.rs:113]: Make sense!

**Pull request merged**